### ISSUE TITLE
ReferenceConfigCache doesn't work well when no provider

### ISSUE THREAD

**Issue by YoungHu** [apache/incubator-dubbo#2233]
- [x] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.6.2 and lower
* Operating System version: all version
* Java version: all version

### Step to reproduce this issue

1. shutdown provider
2. consumer side use `ReferenceConfigCache.get()` to invoke provider
3. invoke provider one time
4. startup provider
5. invoker provider again

Pls. provide [GitHub address] to reproduce this issue.

### Expected Result

when provider startup, the second time invoke should success.

### Actual Result

second time invoke failed

### What cause it
When no provider, `config.get()` return null, at this time we shouldn't store this reference into cache.

```java
public <T> T get(ReferenceConfig<T> referenceConfig) {
        String key = generator.generateKey(referenceConfig);

        ReferenceConfig<?> config = cache.get(key);
        if (config != null) {
            return (T) config.get();
        }

        cache.putIfAbsent(key, referenceConfig);
        config = cache.get(key);
        return (T) config.get();
    }
```

**Comment by diecui1202** [comment]
@YoungHu Could u please send a PR to solve this problem ?

**Comment by YoungHu** [comment]
PR already done

**Comment by diecui1202** [comment]
The same issue: #1765 . But it's not ReferenceConfigCache's problem.

I think we need to fix it like #1999

**Comment by diecui1202** [comment]
@YoungHu Could u refer to #1999 and fix it on dubbo-2.6.x ?

**Comment by YoungHu** [comment]
will check it tomorrow by using the master branch code

**Comment by YoungHu** [comment]
confirmed this issue fix at master branch with #1999 ,close this one
